pyiceberg/catalog/rest.py

```diff
@@ -472,6 +488,7 @@ def create_table(
         table_response = TableResponse(**response.json())
         return self._response_to_table(self.identifier_to_tuple(identifier), table_response)
 
+    @retry(retry=retry_if_exception_type(AuthorizationExpiredError), stop=stop_after_attempt(2), before=_retry_hook, reraise=True)
```

Comment:
Instead of adding the decorators everywhere, maybe we can wrap the logic in `_handle_non_200_response`. The major issue I see with the current approach is:

- The decorater is duplicated many times
- It is easy to miss a certain method, for example, `_do_commit()` is also very likely to time out when you do a write operation that takes some time.

Reply:
Thanks @Fokko for taking time to review this.

- That's a good point and I agree with your intention of keeping the retry logic to a minimal.  I put some thought into how I'd need to implement it if we want to put it in `_handle_non_200_response`, but by the time we reach that method, we've lost the caller and the request that was sent. Hence, the next best thing I could come up was to remove the retry argument duplicates all over and keep it to a minimal. 
- My bad that I missed `_commit_table()` added the functionality there.